pkg/server/test/read_changes.go

```diff
@@ -265,3 +283,34 @@ func setup(store string, datastore storage.OpenFGADatastore) (context.Context, s
 
 	return ctx, datastore, nil
 }
+
+// writeFourTuplesConcurrently to expose potential race issues when reading changes
+func writeFourTuplesConcurrently(t *testing.T, store string, datastore storage.OpenFGADatastore) (context.Context, storage.ChangelogBackend) {
```

Comment:
Instead of harding coding to only write four tuples, perhaps better to accept two groups of tuples keys as parameters.  Each group will be in its own go func?

Reply:
What is the benefit of doing that?

Reply:
It is better to tie the expectation to where the input is... in line 267/272, I have no idea why I should expect 4 or 3 unless I see the implementation of the function here in line 288.  Not a blocker but something to keep in mind to improve readability